File: solutions/fortran/reverse-string/1/reverse_string.f90
module reverse_string
  implicit none
contains

  function reverse(input) result(reversed)
    character(*), intent(in) :: input
    character(len=len(input)) :: reversed
    integer :: i
    i = 0
    do i = len(input), 1, -1
       reversed(len(reversed) - i + 1:len(reversed) - i + 1) = input(i:i)
    end do
  end function

end module

File: solutions/cpp/reverse-string/1/reverse_string.cpp
#include "reverse_string.h"
using namespace std;
#include <string>

namespace reverse_string 
{
string reverse_string(string s)
{
    string result = "";
    for (int i = s.length() - 1; i >= 0; i--) {
        result += s[i];
    }
    return result;
}
}  // namespace reverse_string
